src/gui/qgsmapcanvas.cpp

```diff
@@ -1498,6 +1489,13 @@ void QgsMapCanvas::mouseReleaseEvent( QMouseEvent* e )
         }
         return;
       }
+
+      // Store clicked x & y into current map tool.
+      QgsPoint point = getCoordinateTransform()->toMapCoordinates( e->x(), e->y() );
+
+      mMapTool->setProperty( "click_x", point.x() );
+      mMapTool->setProperty( "click_y", point.y() );
```

Comment:
Could that code be moved to QgsMapToolIdentifyAction? It should have access to mouse coordinates and the canvas, so all the context for this calculation is there. This would avoid always calculating this property and setting it even on tools that don't need it.